socs/agents/ups/agent.py

```diff
@@ -330,21 +331,29 @@ def acq(self, session, params=None):
                     'upsOutputSource']
 
             for oid in oids:
+                main_get_list.append(('UPS-MIB', oid, 0))
                 get_list.append(('UPS-MIB', oid, 0))
 
+            ups_get_result = yield self.snmp.get(get_list, self.version)
+
             # Append input OIDs to GET list
             input_oids = ['upsInputVoltage',
                           'upsInputCurrent',
                           'upsInputTruePower']
 
             # Use number of input lines used to append correct number of input OIDs
-            num_lines = [('UPS-MIB', 'upsInputNumLines', 0)]
-            num_res = yield self.snmp.get(num_lines, self.version)
+            input_num_lines = [('UPS-MIB', 'upsInputNumLines', 0)]
+            num_res = yield self.snmp.get(input_num_lines, self.version)
             if num_res is not None:
```

Comment:
What does it mean, if num_res is None?

I ask because in that case, the "if" block is not executed ... and input_get_results will hold the value it had in the previous loop iteration.  And then, I guess, those values could get sent to the feed, with misleading timestamps.

Same issue for the output_get_results.

Perhaps, if _any_ of these snmp.get return None, a connection error should be processed and reconnect initiated?

Reply:
Good point, if any snmp.get returns None then that means some connection issue is happening, so it should probably go directly to the connection error.